### PR TITLE
Fix image build in release

### PR DESCRIPTION
**Pull request by adrcunha**
`$KO_DOCKER_REPO` must be exported, or `ko` won't see it.

Bonuses:
* make `publish_yaml` more verbose
* log the config before building, for easier debugging/investigation/auditing

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -43,9 +42,14 @@ function tag_images_in_yaml() {
 # If $TAG is not empty, also copy it to $RELEASE_GCS_BUCKET bucket's "previous" directory.
```
Comment (srinivashegde86) [inline scripts/release.sh]: Update documentation

File: scripts/release.sh [hunk 2]
```diff
@@ -174,10 +178,26 @@ function run_validation_tests() {
...
+    echo "- Artifacts WILL NOT be tagged"
+  fi
   if (( PUBLISH_RELEASE )); then
-    echo "- Destination GCS: ${RELEASE_GCS_BUCKET}"
+    echo "- Release WILL BE published to '${RELEASE_GCS_BUCKET}'"
+  else
+    echo "- Release will not be published"
   fi
+  if (( BRANCH_RELEASE )); then
+    echo "- Release WILL BE branched from '${RELEASE_BRANCH}'"
```
Comment (srinivashegde86) [inline scripts/release.sh]: can we add this to the next if condition for branch_release.

File: scripts/release.sh (hunk 1 above)
Comment (adrcunha) [reply scripts/release.sh]: The documentation is up-to-date. What am I missing?

Comment (adrcunha) [reply scripts/release.sh]: Hmmm... Maybe you mean the documentation about `tag_images_in_yaml()`?

File: scripts/release.sh (hunk 2 above)
Comment (adrcunha) [reply scripts/release.sh]: I don't want to, since these are two distinct functional blocks: one for logging purposes only, the other for actions.

Comment (srinivashegde86) [reply scripts/release.sh]: okies.sg.

File: scripts/release.sh (hunk 1 above)
Comment (adrcunha) [reply scripts/release.sh]: Updated the documentation about tag_images_in_yaml().

Comment (srinivashegde86) [reply scripts/release.sh]: Sorry, I meant publish_yaml. It only has $1 param in doc and you are also using $2

Comment (adrcunha) [reply scripts/release.sh]: Read the code again, you missed the internal helper function `verbose_gsutil_cp()`.

Comment (srinivashegde86) [reply scripts/release.sh]: oops. did not realize that it was an internal function

**Pull request merged**